src/Features/CSharp/Portable/TodoComments/CSharpTodoCommentIncrementalAnalyzerProvider.cs

```diff
@@ -5,85 +5,98 @@
 using System.Collections.Immutable;
 using System.Composition;
 using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.Host.Mef;
 using Microsoft.CodeAnalysis.TodoComments;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.CSharp.TodoComments
 {
-    [ExportLanguageService(typeof(ITodoCommentService), LanguageNames.CSharp), Shared]
-    internal class CSharpTodoCommentService : AbstractTodoCommentService
+    [ExportLanguageServiceFactory(typeof(ITodoCommentService), LanguageNames.CSharp), Shared]
+    internal class CSharpTodoCommentServiceFactory : ILanguageServiceFactory
     {
-        private static readonly int s_multilineCommentPostfixLength = "*/".Length;
-        private const string SingleLineCommentPrefix = "//";
+        public ILanguageService CreateLanguageService(HostLanguageServices languageServices)
+        {
+            return new CsharpTodoCommentService(languageServices.WorkspaceServices.Workspace);
```

Comment:
To make the diff easier, can you just make this a sibling class for now and not a nested class?